val.py

```diff
@@ -276,6 +276,11 @@ def run(
         tp, fp, p, r, f1, ap, ap_class = ap_per_class(*stats, plot=plots, save_dir=save_dir, names=names)
         ap50, ap = ap[:, 0], ap.mean(1)  # AP@0.5, AP@0.5:0.95
         mp, mr, map50, map = p.mean(), r.mean(), ap50.mean(), ap.mean()
+
+        available_names = [names[c] for c in ap_class]
+                   
+        callbacks.run('on_val_end_compute_metrics', available_names, ap50, ap, f1, p, r)
```

Comment:
I am gonna add this, but we might take it out if the repo maintainers don't want it -- wondering why no logging callbacks are present in the validation scripts.